RAD-229: Clean up the documentation after all the RAD updates. (#745)
spacetelescope/rad: docs/creating.rst

under headings and subheadings (or even deeper) as you see fit. In particular,
you should have a clear idea of the following:

    #. The base name of the schema you which to create. This should be a short
       descriptive name that should only include lower case letters and underscores.

>        No spaces or hyphens should be used. For example, ``dark``, ``aperture``,

Reviewer: ```suggestion
       No spaces or hyphens should be used in the name or path. For example, ``dark``, ``aperture``,
```

       ``wcsinfo``, ``cal_logs``, etc. Typically, this name will form the base name
       that the object you are describing will be called in Python. For example,
       ``aperture`` may correspond to an ``Aperture`` object in Python. This is not

>        always the case, but it is a good rule of thumb. Form now on we will use

Reviewer: ```suggestion
       always the case, but it is a good rule of thumb. From now on we will use
```

          to create.

    #. Next you will need to decided where in RAD to locate your schema. Look inside
       the ``latest`` directory in the top of your working copy of RAD to see the
       organization of the existing schemas.

       .. note::
        In the end you will have something like: ``latest/<path/to/your/schema>``,
        where ``<path/to/your/schema>`` is the path to the directory inside ``latest``
        that your schema will be located. This means you now have the URI
        fragment ``<path/to/your/schema>/<name>-<version>``, which can be formed
        into the full RAD schema URI by prefixing it with
        ``asdf://stsci.edu/datamodels/roman/schemas/``. E.G.

        .. code::yaml

            asdf://stsci.edu/datamodels/roman/schemas/<path/to/your/schema>/<name>-<version>

       * If your schema is not for the SOC then it should be located within a
         directory that corresponds to the origin of the data. For example, the
         IPAC/SSC schemas are located in ``latest/SSC``. Items located within
         those directories should follow the organisational structure of the

>          schemas already in that directory. These directory names be in ALL CAPS.

Reviewer: ```suggestion
         schemas already in that directory. These directory names must be in ALL CAPS.
```

             ``latest/meta/``. For example, ``latest/meta/exposure.yaml`` is the
             schema for the ``meta.exposure`` attribute of Romancal pipeline outputs.

          #. If your schema provides some enumerated object which is used in multiple
             places, then it should be located in ``latest/enums/``. For example,
             ``latest/enums/wfi_detector.yaml`` is the schema which lists all the
             possible WFI detectors that can be used in Romancal pipeline outputs.

          #. If your schema works to describe the columns of a table, then it should be
             located in ``latest/tables/``. For example, ``latest/tables/prompt_catalog_table.yaml``
             describes the columns for the prompt source catalog table that is created by
             the Romancal pipeline.

          #. If your schema does not fit into one of the above categories, then it should
             be discussed with the RAD maintainers to determine the best location.

    #. The keywords for your fields and their hierarchical organization.

       * *Keywords* are the key in the key-value pairs for information. For example,

>          it may refer to a specific value like ``ra_ref`` for the right ascention

Reviewer: ```suggestion
         the keyword may refer to a specific value like ``ra_ref`` for the right ascension
```

         of the reference position for the WCS. Or it may refer to an entire group

>          of related information like ``wcsinfo`` contains keywords that point at

Reviewer: ```suggestion
         of related information like ``wcsinfo``, which contains keywords that point at
```

        Sometimes you might want to have a field which is required, but which
        may not take on any values at all. In this case you can use the
        ``null`` type as one of the possibilities in the ``anyOf`` combiner.

Add Required and Ordering Information
--------------------------------------

After you have added all of your fields, you will want to add the required
and ordering information. This is done at the same indentation level as the
``properties`` keyword.

.. code:: yaml

    required: [<required field 1>, <required field 2>, <required field 3>]
    propertyOrder: [<field 1>, <field 2>, <field 3>]

.. warning::
    The ``propertyOrder`` can only be included in schemas that are tagged. using
    it outside of a tagged context will cause ASDF to fail to validate things using

>     the schema, that might otherwise be valid.

Reviewer: ```suggestion
    it outside of a tagged context will cause ASDF to fail to validate the schema, even if it might otherwise be valid.
```

.. _tag-your-schema:

Tag Your Schema
---------------

.. warning::
    You should only tag your top-level schema, the one that describes an entire product.

We suggest that when using the :ref:`RAD Helper Tool <rad_helper>` to create your schema, you

> also use it to tag your schema. This can be done via selecting the ``tag`` option to on.

Reviewer: ```suggestion
also use it to tag your schema. This can be done via selecting the ``tag`` option.
```
